test/RibbonThetaSTETHVault.ts

```diff
@@ -0,0 +1,2907 @@
+          const afterBalance = await assetContract.balanceOf(user);
+          actualWithdrawAmount = afterBalance.sub(beforeBalance);
+        }
+
+        // Should be less because the pps is down
+        assert.bnLt(actualWithdrawAmount, depositAmount);
+        assert.bnEqual(actualWithdrawAmount, withdrawAmount);
+      });
+
+      it("fits gas budget [ @skip-on-coverage ]", async function () {
+        await rollToSecondOption(firstOption.strikePrice);
+
+        const tx = await vault.completeWithdraw(minETHOut, { gasPrice });
+        const receipt = await tx.wait();
+
+        assert.isAtMost(receipt.gasUsed.toNumber(), 159946);
+        // console.log(
+        //   params.name,
+        //   "completeWithdraw",
+        //   receipt.gasUsed.toNumber()
+        // );
+      });
+    });
+
+    describe("#setStrikePrice", () => {
+      time.revertToSnapshotAfterEach();
+
+      it("should revert if not owner", async function () {
+        await expect(
+          vault.connect(userSigner).setStrikePrice(parseEther("10"))
+        ).to.be.revertedWith("Ownable: caller is not the owner");
+      });
+
+      it("should set the new strike price", async function () {
+        await vault.connect(ownerSigner).setStrikePrice(parseEther("10"));
+        assert.bnEqual(
+          BigNumber.from(await vault.overriddenStrikePrice()),
+          parseEther("10")
+        );
+      });
+    });
+
+    describe("#setCap", () => {
+      time.revertToSnapshotAfterEach();
+
+      it("should revert if not owner", async function () {
+        await expect(
+          vault.connect(userSigner).setCap(parseEther("10"))
+        ).to.be.revertedWith("Ownable: caller is not the owner");
+      });
+
+      it("should set the new cap", async function () {
+        await vault.connect(ownerSigner).setCap(parseEther("10"));
+        assert.equal((await vault.cap()).toString(), parseEther("10"));
+      });
+
+      it("should revert when depositing over the cap", async function () {
+        const capAmount = BigNumber.from("100000000");
+        const depositAmount = BigNumber.from("10000000000");
+        await vault.connect(ownerSigner).setCap(capAmount);
+
+        await expect(
+          vault.depositETH({ value: depositAmount })
+        ).to.be.revertedWith("Exceed cap");
+      });
+    });
+
+    describe("#sendLDORewards", () => {
+      time.revertToSnapshotAfterEach();
+
+      it("should send LDO rewards to feeRecipient", async function () {
+        const LDO_HOLDER = "0xF5Fb27b912D987B5b6e02A1B1BE0C1F0740E2c6f";
+        await hre.network.provider.request({
+          method: "hardhat_impersonateAccount",
+          params: [LDO_HOLDER],
+        });
+        const ldoHolder = await ethers.provider.getSigner(LDO_HOLDER);
+
+        const ldo = await ethers.getContractAt("IERC20", LDO_ADDRESS);
+
+        let ldoDepositAmount = BigNumber.from(100);
+
+        await ldo.connect(ldoHolder).transfer(vault.address, ldoDepositAmount);
```

Comment:
transaction reverts on .transfer() of ldo tokens. do you see offhand anything I am doing wrong? EOA has ldo and has eth

Reply:
It could be related to the LDO vesting. Check out the [doTransfer function in LDO](https://etherscan.io/address/0x5a98fcbea516cf06857215779fd812ca3bef1b32#code), which checks the [Controller](https://etherscan.io/address/0xf73a1260d222f447210581ddf212d915c09a3249) if an account can spend their LDOs or not.

<img width="420" alt="Screenshot 2021-08-19 at 10 22 27 PM" src="https://user-images.githubusercontent.com/5748750/130138626-74c598a8-cf68-46d6-a92a-c454fea88e57.png">

Try getting a smaller LDO retail holder

Reply:
this was it!